#! /bin/sh /usr/share/dpatch/dpatch-run
## 80_zip.dpatch by Mike Hommey <glandium@debian.org>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Avoid needing zip if not required. bz#331785.

@DPATCH@

=== configure.in
==================================================================
--- xulrunner/configure.in	(revision 42)
+++ xulrunner/configure.in	(local)
@@ -518,10 +518,6 @@
 AC_PATH_PROG(WHOAMI, whoami, :)
 AC_PATH_PROG(AUTOCONF, autoconf, :)
 AC_PATH_PROG(UNZIP, unzip, :)
-AC_PATH_PROGS(ZIP, zip)
-if test -z "$ZIP" || test "$ZIP" = ":"; then
-    AC_MSG_ERROR([zip not found in \$PATH])
-fi
 AC_PATH_PROG(SYSTEM_MAKEDEPEND, makedepend)
 AC_PATH_PROG(XARGS, xargs)
 if test -z "$XARGS" || test "$XARGS" = ":"; then
@@ -5660,6 +5656,14 @@
     AC_MSG_ERROR([--enable-chrome-format must be set to either jar, flat, both, or symlink])
 fi
 
+if test "$MOZ_CHROME_FILE_FORMAT" = "jar" ||
+    test "$MOZ_CHROME_FILE_FORMAT" = "both"; then
+    AC_PATH_PROGS(ZIP, zip)
+    if test -z "$ZIP" || test "$ZIP" = ":"; then
+        AC_MSG_ERROR([zip not found in \$PATH])
+    fi
+fi
+
 dnl ========================================================
 dnl = Define default location for MOZILLA_FIVE_HOME
 dnl ========================================================
=== config/config.mk
==================================================================
--- xulrunner/config/config.mk	(revision 42)
+++ xulrunner/config/config.mk	(local)
@@ -475,10 +475,14 @@
 # Flags passed to make-jars.pl
 
 MAKE_JARS_FLAGS = \
-	-s $(srcdir) -t $(topsrcdir) -z $(ZIP) -p $(MOZILLA_DIR)/config/preprocessor.pl \
+	-s $(srcdir) -t $(topsrcdir) -p $(MOZILLA_DIR)/config/preprocessor.pl \
 	-f $(MOZ_CHROME_FILE_FORMAT) \
 	$(NULL)
 
+ifdef ZIP
+MAKE_JARS_FLAGS += -z $(ZIP)
+endif
+
 ifdef NO_JAR_AUTO_REG
 MAKE_JARS_FLAGS += -a
 endif
=== config/make-jars.pl
==================================================================
--- xulrunner/config/make-jars.pl	(revision 42)
+++ xulrunner/config/make-jars.pl	(local)
@@ -137,7 +137,7 @@
     $zipprog = $::opt_z;
 }
 
-if ($zipprog eq "") {
+if (($fileformat eq "jar" || $fileformat eq "both") && $zipprog eq "") {
     print "A valid zip program must be given via the -z option or the ZIP environment variable. Exiting.\n";
     exit(1);
 }
@@ -509,6 +509,7 @@
                 my $dest = $1;
                 my $srcPath = defined($2) ? substr($2, 1, -1) : $2;
                 EnsureFileInDir("$chromeDir/$jarfile", $baseFilesDir, $dest, $srcPath, 0, 0);
+                EnsureFileInDir("$jarDir/$jarfile", $baseFilesDir, $dest, $srcPath, 0, 0) if ($fileformat eq "flat");
                 $args = "$args$dest ";
                 if (!foreignPlatformFile($jarfile)  && $autoreg &&
                     $dest =~ /([\w\d.\-\_\+]+)\/([\w\d.\-\_\\\/]+)contents.rdf/)
@@ -521,6 +522,7 @@
                 my $dest = $1;
                 my $srcPath = defined($2) ? substr($2, 1, -1) : $2;
                 EnsureFileInDir("$chromeDir/$jarfile", $baseFilesDir, $dest, $srcPath, 1, 0);
+                EnsureFileInDir("$jarDir/$jarfile", $baseFilesDir, $dest, $srcPath, 1, 0) if ($fileformat eq "flat");
                 $overrides = "$overrides$dest ";
                 if (!foreignPlatformFile($jarfile)  && $autoreg && $dest =~ /([\w\d.\-\_\+]+)\/([\w\d.\-\_\\\/]+)contents.rdf/)
                 {
@@ -533,6 +535,7 @@
                 my $dest = $1;
                 my $srcPath = defined($2) ? substr($2, 1, -1) : $2;
                 EnsureFileInDir("$chromeDir/$jarfile", $baseFilesDir, $dest, $srcPath, 1, 1);
+                EnsureFileInDir("$jarDir/$jarfile", $baseFilesDir, $dest, $srcPath, 1, 1) if ($fileformat eq "flat");
                 $overrides = "$overrides$dest ";
                 if (!foreignPlatformFile($jarfile)  && $autoreg && $dest =~ /([\w\d.\-\_\+]+)\/([\w\d.\-\_\\\/]+)contents.rdf/)
                 {
